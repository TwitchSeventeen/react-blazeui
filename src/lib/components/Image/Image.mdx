---
name: Image
route: /objects/image
menu: Objects
---

import { Playground, PropsTable } from 'docz'
import Image from './index';

# Code

Used to display a snippet of code.

## Basic usage
By default the code snippet is displayed inline.
<Playground>
    <Image />
</Playground>


## Properties
<PropsTable of={Image} />