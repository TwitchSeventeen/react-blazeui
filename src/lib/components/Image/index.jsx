import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Image(props) {
  const {
    className, src, alt, user, filter, likes, height, width, ...other
  } = props;
  const classes = classNames(
    'o-image',
    className,
  );

  let source = src;

  if (!src) {
    const userProp = user ? `user/${user}/` : '';
    const likesProp = likes ? 'likes/' : '';
    const filterProp = filter ? `${(height && width) ? '' : 'featured/'}?${filter}` : '';
    const sizeProp = (height && width) ? `${width}x${height}/` : '';
    source = `https://source.unsplash.com/${filter || user ? '' : 'random/'}${userProp}${sizeProp}${likesProp}${filterProp}`;
  }

  return (
    <img {...other} className={classes} src={source} alt={alt} />
  );
}

Image.propTypes = {
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Url for the image
   */
  src: PropTypes.string,
  /**
   * Alternate text
   */
  alt: PropTypes.string,
  /**
   * Choose a random photo from a specific user.
   * From Unsplash.
   */
  user: PropTypes.string,
  /**
   * Choose a random photo that's been liked by a specific user.
   * Use with the user prop.
   */
  likes: PropTypes.bool,
  /**
   * Comma separated list of filters to find images by
   */
  filter: PropTypes.string,
  /**
   * Height of the image to fetch
   */
  height: PropTypes.number,
  /**
   * Width of the image to fetch
   */
  width: PropTypes.number,
};

Image.defaultProps = {
  className: null,
  src: null,
  alt: 'random placeholder image',
  user: null,
  likes: false,
  filter: null,
  height: null,
  width: null,
};

export default Image;
