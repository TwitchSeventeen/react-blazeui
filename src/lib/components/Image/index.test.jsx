import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Image from './index';

describe('Image', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Image).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Image
        height={100}
        width={50}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly without a width', () => {
    const { asFragment } = render(
      <Image
        height={100}
        filter="lorem"
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with dimensions and a filter', () => {
    const { asFragment } = render(
      <Image
        height={100}
        width={50}
        filter="lorem"
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with a picture liked by a user', () => {
    const { asFragment } = render(
      <Image
        user="twitchseventeen"
        filter="lorem"
        likes
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with a explicit picture', () => {
    const { asFragment } = render(
      <Image
        src="https://via.placeholder.com/150"
        alt="alt text"
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
