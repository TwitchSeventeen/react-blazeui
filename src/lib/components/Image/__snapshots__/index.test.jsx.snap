// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly 1`] = `
<img
  alt="random placeholder image"
  className="o-image"
  src="https://source.unsplash.com/random/50x100/"
/>
`;

exports[`Image renders correctly with a explicit picture 1`] = `
<img
  alt="alt text"
  className="o-image"
  src="https://via.placeholder.com/150"
/>
`;

exports[`Image renders correctly with a picture liked by a user 1`] = `
<img
  alt="random placeholder image"
  className="o-image"
  src="https://source.unsplash.com/user/twitchseventeen/likes/featured/?lorem"
/>
`;
